tests/rados/test_mgr_daemon.py

```diff
@@ -0,0 +1,134 @@
+    mgr_nodes = ceph_cluster.get_nodes(role="mgr")
+    mgr_obj = MgrWorkflows(node=cephadm)
+
+    log.info("Scenario 1: Verification of MGR fail scenario")
+    # Get the active MGR name
+    mgr_stats = mgr_obj.get_mgr_stats()
+    active_mgr = mgr_stats["active_name"]
+    # Make active mgr fail
+    mgr_obj.set_mgr_fail(active_mgr)
+    # Check the new mgr is activated
+    mgr_stats = mgr_obj.get_mgr_stats()
+    new_active_mgr = mgr_stats["active_name"]
+
+    if new_active_mgr is active_mgr:
+        log.error("After the MGR failure cluster ")
+        return 1
+    log.info(f"The new active manager is -{new_active_mgr}")
+    # Check the ceph status
+    rados_obj.log_cluster_health()
+    log.info("End of Scenario-1: Verification of MGR fail scenario passed")
+
+    log.info("Scenario 2: Verification of MGR restart scenario")
+    # Restart MGR and check
+    cephadm.shell(["ceph orch restart mgr"])
+    time.sleep(10)
+    rados_obj.log_cluster_health()
+    log.info("End of Scenario 2: Verification of MGR restart scenario passed")
+
+    log.info(
+        "Scenario 3: Verification of  a scenario of adding and removing mgr daemon"
+    )
+    # Adding a new MGR
+    new_node = get_node(ceph_cluster)
+    if new_node is None:
+        log.error("No new node to add the MGR daemon")
+        return 1
+    result = mgr_obj.add_mgr_service(host=new_node)
+    if not result:
+        log.error("Not able to add the mgr daemon to the cluster")
+        return 1
+
+    result = mgr_obj.remove_mgr_service(host=new_node)
+    if not result:
+        log.error("Not able to remove the mgr daemon to the cluster")
+        return 1
+    log.info(
+        "End of Scenario 3: Verification of  a scenario of adding and removing mgr daemon passed"
+    )
+
+    log.info(
+        "Scenario 4: Verification of  a scenario of scaleup and down of mgr daemon"
+    )
+    # Get the MGR count in the cluster
+    mgr_nodes = ceph_cluster.get_nodes(role="mgr")
+    new_list = []
+    for node in mgr_nodes:
+        new_list.append(node.hostname)
+    new_node = get_node(ceph_cluster)
+    new_list.append(new_node.hostname)
+    mgr_count = get_mgr_count(ceph_cluster)
+    log.info("The mgr count before scaleup is - {mgr_count}")
+
+    result = mgr_obj.mgr_scale_up_down(new_list)
+    if not result:
+        log.error("Not able to scale-up the mgr daemon to the cluster")
+        return 1
+    mgr_count = get_mgr_count(ceph_cluster)
+    log.info(f"The mgr count after scaleup is - {mgr_count}")
+    new_list.pop()
+    # scale down scenario
+    mgr_obj.mgr_scale_up_down(new_list)
+    mgr_count = get_mgr_count(ceph_cluster)
+    if not result:
+        log.error("Not able to scale-down the mgr daemon to the cluster")
+        return 1
+    log.info(f"The mgr count after scale-down is - {mgr_count}")
+    log.info(
+        "End of Scenario 4: Verification of  a scenario of scaleup and down of mgr daemon passed"
+    )
+    return 0
+
+
+def get_node(cluster):
```

Comment:
The method name is ambiguous.  please update the method name to something that would indicate that we are requesting for a node on host without mon daemon deployed.

Reply:
Modified the method name